src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -5833,9 +5835,10 @@ private bool IsUsingAliasInScope(string name)
             }
         }
 
-        private static void WarnOnAccessOfOffDefault(SyntaxNode node, BoundExpression boundLeft, DiagnosticBag diagnostics)
+        private void WarnOnAccessOfOffDefault(SyntaxNode node, BoundExpression boundLeft, DiagnosticBag diagnostics)
         {
-            if (boundLeft != null && boundLeft.Kind == BoundKind.DefaultExpression && boundLeft.ConstantValue == ConstantValue.Null)
+            if (boundLeft != null && boundLeft.Kind == BoundKind.DefaultExpression && boundLeft.ConstantValue == ConstantValue.Null &&
+                Compilation.LanguageVersion < MessageID.IDS_FeatureNullableReferenceTypes.RequiredVersion())
```

Comment:
I'm not sure that this is safe. We need to make sure that NRT is actually _turned on_ at this point in the code, or we've just introduced a hole.

Reply:
I don't care.  We're removing a warning that we reported in previous versions.  The warning wasn't required by any specification, we just reported it because we were being nice.  Because it wasn't required, it isn't a hole.  In C# 8 we are introducing a more comprehensive system for getting null warnings if you want them, and removing the older warning.  We could just delete that warning from the code base altogether, but I thought it would be nice to retain it for language versions less than 8.


---
In reply to: [276816873](https://github.com/dotnet/roslyn/pull/34889#discussion_r276816873) [](ancestors = 276816873)